### PR TITLE
Adds in functionality for lockboxes

### LINKED ISSUE #2336: Add lockboxes
# Description

Lockboxes are similar to storage boxes, with the added feature that they can be locked. They are also bulky instead of normally sized. Lockboxes can be locked and unlocked with an appropriate id card. By using an emag on one, they are broken open and id cards will no longer work. Lockboxes are used to store items of value at roundstart such as clusterbangs or mindshield implants. Additionally, a medal variant exists that is given to a few departments, most notably to the captain which contains a Medal of Captaincy, a possible item for traitors to purloin.

https://github.com/tgstation/tgstation/blob/547384efaa5864bddf03c3baa871ceaa0ac5d3d6/code/game/objects/items/storage/lockbox.dm

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #2336

CL: [New] Added in functionality for lockboxes

Plug'n'play script so can be added onto other bags as well.

Also created two variants for lockboxes, one with security access and one with captain only.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
forgot to add the new prefabs to the network manager

**Comment by MaxIsJoe** [comment]
fixed

File: UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			{
+				isLocked = false;
+				isEmagged = true;
+				spriteHandler.SetSpriteSO(unlockedSprite);
+				SparkUtil.TrySpark(interaction.Performer);
+				return;
+			}
+			isLocked = !isLocked;
+			spriteHandler.SetSpriteSO(isLocked ? lockedSprite : unlockedSprite);
+			Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} peeps as it accepts this card.");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs]:
```suggestion
			Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} beeps as it accepts this card.");
```

File: UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		}
+
+		public void ServerPerformInteraction(InventoryApply interaction)
+		{
+			if (interaction.UsedObject != null)
+			{
+				if (interaction.UsedObject.TryGetComponent<IDCard>(out var card) && card.HasAccess(allowedAccess) == false)
+				{
+					Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} peeps as it refuses access from this card.");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs]:
```suggestion
					Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} beeps as it refuses access from this card.");
```

File: UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+					return;
+				}
+				if(card != null && card.HasAccess(allowedAccess))
+				{
+					isLocked = !isLocked;
+					spriteHandler.SetSpriteSO(isLocked ? lockedSprite : unlockedSprite);
+					Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} peeps as it accepts this card.");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs]:
```suggestion
					Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} beeps as it accepts this card.");
```

File: UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			return DefaultWillInteract.Default(interaction, side) != false && interaction.UsedObject != null && isEmagged == false;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if(interaction.UsedObject.TryGetComponent<IDCard>(out var card) == false ||
+			   interaction.UsedObject.TryGetComponent<Emag>(out var mag)) return;
+			if (card != null && card.HasAccess(allowedAccess) == false)
+			{
+				Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} peeps as it refuses access from this card.");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Storage/Lockbox.cs]:
```suggestion
				Chat.AddExamineMsg(interaction.Performer, $"The {gameObject.ExpensiveName()} beeps as it refuses access from this card.");
```

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**